source/spirv_validator_options.h

```diff
@@ -0,0 +1,56 @@
+// Copyright (c) 2017 Google Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef LIBSPIRV_SPIRV_VALIDATOR_OPTIONS_H_
+#define LIBSPIRV_SPIRV_VALIDATOR_OPTIONS_H_
+
+#include "spirv-tools/libspirv.h"
+
+// Return true if the command line option for the validator limit is valid (Also
+// returns the Enum for option in this case). Returns false otherwise.
+bool spvParseUniversalLimitsOptions(const char* s, spv_validator_limit* limit);
+
+struct validator_universal_limits_t {
+  uint32_t max_struct_members;
+  uint32_t max_struct_depth;
+  uint32_t max_local_variables;
+  uint32_t max_global_variables;
+  uint32_t max_switch_branches;
+  uint32_t max_function_args;
+  uint32_t max_control_flow_nesting_depth;
+  uint32_t max_access_chain_indexes;
+};
+
+const validator_universal_limits_t kDefaultValidatorUniversalLimits = {
```

Comment:
I recommend you use default member initialization instead:
http://en.cppreference.com/w/cpp/language/data_members

That would possibly be more robust to future changes, is more readable, and is shorter.  You'd just need a comment toward the top saying the default initialization is to the default universal limit.